pkg/operator/podidentity/podidentitywebhook_controller_test.go

```diff
@@ -0,0 +1,317 @@
+/*
+Copyright 2018 The OpenShift Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package podidentity
+
+import (
+	"context"
+	"fmt"
+	clientgotesting "k8s.io/client-go/testing"
+	"strings"
+	"testing"
+
+	log "github.com/sirupsen/logrus"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/golang/mock/gomock"
+	admissionv1 "k8s.io/api/admissionregistration/v1"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	policyv1 "k8s.io/api/policy/v1"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/kubernetes"
+	fakeclientgo "k8s.io/client-go/kubernetes/fake"
+	"k8s.io/client-go/kubernetes/scheme"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+
+	configv1 "github.com/openshift/api/config/v1"
+	schemeutils "github.com/openshift/cloud-credential-operator/pkg/util"
+	"github.com/openshift/library-go/pkg/operator/events"
+	"github.com/openshift/library-go/pkg/operator/resource/resourceapply"
+)
+
+func init() {
+	log.SetLevel(log.DebugLevel)
+}
+
+func TestPodIdentityWebhookController(t *testing.T) {
+	schemeutils.SetupScheme(scheme.Scheme)
+
+	getDeployment := func(c kubernetes.Interface, name, namespace string) *appsv1.Deployment {
+		deployment, err := c.AppsV1().Deployments(namespace).Get(context.TODO(), name, v1.GetOptions{})
+		if err == nil {
+			return deployment
+		}
+		return nil
+	}
+
+	getPDB := func(c kubernetes.Interface, name, namespace string) *policyv1.PodDisruptionBudget {
+		pdb, err := c.PolicyV1().PodDisruptionBudgets(namespace).Get(context.TODO(), name, v1.GetOptions{})
+		if err == nil {
+			return pdb
+		}
+		return nil
+	}
+
+	getWebhook := func(c kubernetes.Interface, name string) *admissionv1.MutatingWebhookConfiguration {
+		webhook, err := c.AdmissionregistrationV1().MutatingWebhookConfigurations().Get(context.TODO(), name, v1.GetOptions{})
+		if err == nil {
+			return webhook
+		}
+		l, err := c.AdmissionregistrationV1().MutatingWebhookConfigurations().List(context.TODO(), v1.ListOptions{})
+		fmt.Println("BBBBBBBBBBB", l, err)
+		fmt.Println("AAAAAAAAAAAAAAAA")
+		return nil
+	}
+
+	t.Setenv("AWS_POD_IDENTITY_WEBHOOK_IMAGE", "aws_identity_image")
```

Comment:
Can we read these at startup and pass them in to the controller as parameters? `os.Setenv` and `os.Getenv` are not thread-safe and while it's not an issue in this code right now it's close to a sharp edge case, if we read at startup we can be more confident in reasoning about the downstream code.

Reply:
Setenv is used only for testing so i don't think we will have any problems with Getenv here.